### PR TITLE
feat(resolve): resolve html refactor

### PR DESCRIPTION
**Pull request by ddivad**
## feat(resolve): resolve html refactor

Refactor HTML resolving code in claws to fix consistency between `scrape` and `result` events. IP Locked providers should now return missing information such as `source`, `quality` and HEAD request information (if available).

### What's in this PR:
- Refactor `resolveHtml.js` to remove the `if, else` structure and replace with a class based structure found in `HTMLResolvers.s` to simplify code structure.
- Redo scrape event handling to re-use existing functionality of `result` events for scrape results, instead of the `scrapeResults` event. Also re-use `wsWrapper` logic to send the events.

# Note:
- This PR will depend on changes in Kamino to 1) remove the `scrapeResult` event handling and 2) Re-order the `source` and `provider` (as they were the wrong way around, which I discovered in testing

### CODE REVIEW

**Comment by NBTX** [comment]
Valid with latest Kamino code.
Still investigating issues regarding scrape results.

File: src/api/resolveLinks.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ const resolveLinks = async (data, ws, req) => {
             for (const result of results) {
                 await wsWrapper.send(result)
             }
+            ws.send(`{"event": "scrapeEnd", "scrapeId": "${data.scrapeId}}"`);
         } catch (err) {
             ws.send(`{"event": "result", "error": "${(err.message || err.toString()).substring(0, 100) + '...'}", "isResultOfScrape": "true"}`);
```
Comment (SpajicM) [inline src/api/resolveLinks.js]: think we should pass scrapeId here as well, so if it fails and Kamino gets an error, we know which one to remove from the pending array.

Comment (NBTX) [reply src/api/resolveLinks.js]: Agreed

Comment (ddivad) [reply src/api/resolveLinks.js]: thanks, good spot :)

File: src/api/resolveLinks.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ const resolveLinks = async (data, ws, req) => {
             for (const result of results) {
                 await wsWrapper.send(result)
             }
+            ws.send(`{"event": "scrapeEnd", "scrapeId": "${data.scrapeId}}"`);
```
Comment (NBTX) [inline src/api/resolveLinks.js]: You put an extra closing bracket after the scrapeId

**Pull request merged**